pkg/apis/pipeline/errors/errors.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2023 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package errors
+
+const userErrorLabel = "[User error] "
+
+type UserError struct {
```

Comment:
I like this approach. I think we can annotate wraps errors into UserErrors and perhaps in the future emit metrics around user vs system errors. 
The main concern I have is that right now we are modifying the `reason` string and we may end up breaking users comparing against the reason string to check the kind of status failure. What do you think of adding the [user error] bit to the message and not the reason?
cc @Yongxuanzhang

Reply:
also relevant the discussion in https://github.com/tektoncd/pipeline/pull/7390#issuecomment-1850663790

Reply:
Yeah, I think adding to message might be safer

Reply:
Thanks both. Updated to label error messages instead.